### ISSUE TITLE
Service launched with vocab in folder makes core dump

### ISSUE THREAD

**Issue by YaYaB** [beniz/deepdetect#399]
I have an issue when I launch a service on an existing folder where there is a file `vocab.dat`

#### Configuration
- Version of DeepDetect:
    - Locally compiled on Ubuntu 16.04 LTS
- Commit (shown by the server when starting):
    - Branch `Master`, 3d2b9eb8b306c587704cff777d52f723dc642b1a
- GPUS:
    - 1 x Nvidia GTX Titan X (Maxwell)

#### Your question / the problem you're facing:
After I've trained a model based on the connector svm I want to make a prediction. However when I launch the service using this model as repository I obtain a core dump. This is due to the presence of the connector `svm`. Any other connector seems to work.

First I launch DeepDetect
INPUT
```
INPUT
./dede --port 8080
```
OUTPUT
```
DeepDetect [ commit 3d2b9eb8b306c587704cff777d52f723dc642b1a ]
[2018-04-20 07:20:26.685] [api] [info] Running DeepDetect HTTP server on localhost:8080
```

Second, I create the service that I name svm_test.
INPUT
```
curl -X PUT 'http://localhost:8080/services/svm_test' -d '{"mllib":"caffe","description":"svm_test classifier","type":"supervised","parameters":{"input":{"connector":"svm"},"mllib":{"nclasses":2}},"model":{"repository":"/home/svm_test/dd_models/model_svm"}}'
```

I obtain the following output
OUTPUT
```
curl: (52) Empty reply from server
```
and
```
[1]    4331 segmentation fault (core dumped)  ./main/dede
```

Now if I remove the `vocab.dat` from the folder it successfully load the model
```
[2018-04-20 07:24:44.387] [api] [info] Running DeepDetect HTTP server on localhost:8080
[2018-04-20 07:24:46.254] [svm_test] [error] BOUM
[2018-04-20 07:24:47.378] [caffe] [info] Initializing net from parameters: 
[2018-04-20 07:24:47.378] [caffe] [info] Creating layer / name=inputl / type=MemorySparseData
[2018-04-20 07:24:47.378] [caffe] [info] Creating Layer inputl
[2018-04-20 07:24:47.378] [caffe] [info] inputl -> data
[2018-04-20 07:24:47.379] [caffe] [info] inputl -> label
....
[2018-04-20 07:24:48.319] [api] [info] 127.0.0.1 "PUT /services/svm_test" 201 2065
```

But then without the `vocab.dat` I can not make any prediction.

Another way to test is to just create a service using a template and then add a vocab.dat to the folder and launch the service again.

Create the service
INPUT
```
curl -X PUT "http://localhost:8080/services/myserv" -d "{\"mllib\":\"caffe\",\"description\":\"example classification service\",\"type\":\"supervised\",\"parameters\":{\"input\":{\"connector\":\"svm\"},\"mllib\":{\"template\":\"mlp\",\"nclasses\":9,\"layers\":[512,512,512],\"activation\":\"prelu\"}},\"model\":{\"repository\":\"/home/svm_test/test\"}}"
```

OUTPUT
```
[2018-04-20 07:27:41.314] [myserv] [info] instantiating model template mlp
[2018-04-20 07:27:41.314] [myserv] [info] source=templates/caffe//mlp/
[2018-04-20 07:27:41.314] [myserv] [info] dest=/home/svm_test/test/mlp.prototxt
[2018-04-20 07:27:41.317] [api] [info] 127.0.0.1 "PUT /services/myserv" 201 1160
```

Then simply create an empty file `vocab.dat` in the folder
```
touch /home/svm_test/test/vocab.dat
```

And launch the service again using the following command
INPUT
```
curl -X PUT "http://localhost:8080/services/myserv" -d "{\"mllib\":\"caffe\",\"description\":\"example classification service\",\"type\":\"supervised\",\"parameters\":{\"input\":{\"connector\":\"svm\"},\"mllib\":{\"nclasses\":9}},\"model\":{\"repository\":\"/home/svm_test/test\"}}" 
```

OUTPUT
```
[1]    4674 segmentation fault (core dumped)  ./main/dede
```

I have taken a look  at the commits. And everything was working well until this commit *06de88ec25e74f3db44cf9b6af858bad9897c19d*

Then it fails for the commit
*fcedd40761621e12faca58790856d3fe9767c581*

I've taken a look at the different version of Caffe, it may be due to the new logs.

There are 9 commits different in those version of Caffe
https://github.com/fantes/caffe/commits/master
https://github.com/fantes/caffe/commits/multi_label_soft

They are some additional content in master between those two commits
- *4ff7f1a1e29faf0bed69b3050ae4c27503dc949f* 

- *359fc2e379cb0e6a4dbe90804ce3c25e88521ef6*

**Comment by beniz** [comment]
OK thanks for the thorough report. Can you run through the crash with gdb:
```
libtool --mode=execute gdb ./dede
```
Then after the crash, use `bt` to get the backtrace. You can report it in the gist, that would be very useful.

**Comment by YaYaB** [comment]
[gdb.txt](https://github.com/beniz/deepdetect/files/1931641/gdb.txt)

**Comment by beniz** [comment]
Thanks for the thorough report, this should be fixed by #400. All unit tests were still passing, so this means our tests are not thorough enough.

**Comment by YaYaB** [comment]
I confirm that it has resolved the issue on my side
